File: bitirme_2/jrclass/src/app/ogrenci-giris/lesson.ts
import { Injectable } from "@angular/core";
import { Ogrenci } from "./ogrenci";
import { Semester } from "./Semester";

@Injectable({
    providedIn: 'root'
  })
export class Lesson{
    lessonID!:number;
    lessonName!:string;
    studentID!:number;
    Ogrenci!:Ogrenci; 
  Semester!: Semester;
   
}

File: BitirmeAPI/BitirmeAPI/Model/Lesson.cs
using System.Text.Json;
using System.Text.Json.Serialization;

namespace BitirmeAPI.Model
{
    public class Lesson
    {
        public int lessonID { get; set; }
        public string? lessonName { get; set; }
        
       
        public Student? Student { get; set; }       
        public int studentID { get; set; }           
        public Note Note { get; set; }
        [JsonIgnore]
        public Teacher Teacher { get; set; }
        public int teacherID { get; set; }
        public Semester Semester { get; set; }
        public int semesterID { get; set; }



    }
}
